src/typechecker.c

```diff
@@ -813,7 +813,14 @@ NODISCARD bool typecheck_expression(AST *ast, Node *expr) {
       /// If there is an initialiser, then its type must match the type of the variable.
       if (expr->declaration.init) {
         if (!typecheck_expression(ast, expr->declaration.init)) return false;
-        if (!convertible(expr->type, expr->declaration.init->type))
+        // Type inference :^)
+        if (!expr->type) {
+          expr->type = expr->declaration.init->type;
+          // TODO: Maybe this shouldn't happen here? A pass after the
+          // bulk of sema that reduces all leftover integer_literals to
+          // integers may be a better choice.
+          if (expr->type == t_integer_literal) expr->type = t_integer;
```

Comment:
I mean, if you’re *inferring* the type, is there ever a situation where preserving the type as `integer_literal` would be of any use?

Reply:
Probably not, to be honest. I was just thinking that maybe this decl would be better suited as a `byte` or some other convertible integer type based on a later use. You're right that I can probably just remove this todo